doc/api/report.md

```diff
@@ -0,0 +1,489 @@
+ an error.
+
+* `--diagnostic-report-uncaught-exception` Enables report to be generated on
+un-caught exceptions. Useful when inspecting JavaScript stack in conjunction
+with native stack and other runtime environment data.
+
+* `--diagnostic-report-on-signal` Enables report to be generated upon receiving
+the specified (or predefined) signal to the running Node process. (See below
+on how to modify the signal that triggers the report). Default signal is `SIGUSR2`.
+Useful when a report needs to be triggered from another program.
+Application monitors may leverage this feature to collect report at regular
+intervals and plot rich set of internal runtime data to their views.
+
+Signal based report generation is not supported in Windows.
+
+Under normal circumstances, there is no need to modify the report triggering
+signal. However, if `SIGUSR2` is already used for other purposes, then this
+flag helps to change the signal for report generation and preserve the original
+meaning of `SIGUSR2` for the said purposes.
+
+* `--diagnostic-report-on-fatalerror` Enables the report to be triggered on
+fatal errors (internal errors within the Node.js runtime, such as out of memory)
+that leads to termination of the application. Useful to inspect various
+diagnostic data elements such as heap, stack, event loop state, resource
+consumption etc. to reason about the fatal error.
+
+* `--diagnostic-report-directory` Location at which the report will be
+generated.
+
+* `--diagnostic-report-filename` Name of the file to which the report will be
+written.
+
+* `--diagnostic-report-signal` Sets or resets the signal for report generation
+(not supported on windows). Default signal is `SIGUSR2`.
+
+* `--diagnostic-report-verbose` Flag that enables additional information to be
+printed during report generation.
+
+A report can also be triggered via an API call from a JavaScript application:
+
+```js
+process.report.triggerReport();
+```
+
+This function takes an optional additional argument `filename`, which is
+the name of a file into which the report is written.
+
+```js
+process.report.triggerReport('./foo.json');
+```
+
+This function takes an optional additional argument `err` - an Error object
+that will be used as the context for the JavaScript stack printed in the report.
+When using report to handle errors in a callback or an exception handler, this
+allows the report to include the location of the original error as well
+as where it was handled.
+
+```js
+try {
+  process.chdir('/non-existent-path');
+} catch (err) {
+  process.report.triggerReport(err);
+}
+// Any other code
+```
+
+If both filename and error object are passed to `triggerReport()` the
+error object must be the second parameter.
+
+```js
+try {
+  process.chdir('/non-existent-path');
+} catch (err) {
+  process.report.triggerReport(filename, err);
+}
+// Any other code
+```
+
+The content of the diagnostic report can be returned as a JSON-formatted object
```

Comment:
The description and the example seem a bit confusing. The signature in the `process.md` states that this function returns object. But the example below names the result `str`. And "JSON-formatted object" seems also murky. Should this be something like "serializable object" or "JSON-compatible object"? And then we can use something like `console.log(JSON.stringify(result);` in the example.